qccalculator/cli.py

```diff
@@ -11,151 +10,252 @@
+                                  version="0.1.0",
+                                  uri="https://github.com/HUPO-PSI/qcML-development/blob/master/cv/v0_1_0/qc-cv.obo")
+  cv_ms = qc.ControlledVocabulary(ref="MS",
+                                  name="Proteomics Standards Initiative Mass Spectrometry Ontology",
+                                  version="4.1.7", uri="https://github.com/HUPO-PSI/psi-ms-CV/blob/master/psi-ms.obo")
+
+  return qc.MzQcFile(version="0.1.0",
+                     creationDate=datetime.now().isoformat(),
+                     runQualities=rqs,
+                     setQualities=sqs,
+                     controlledVocabularies=[cv_qc, cv_ms])
+
+
+# @click.pass_context
+def finale():
+  """
+  Gunzip and compress the output files.
+
+  Returns
+  -------
+  """
+  logging.warn("Calculated metrics from {} different input peak files".format(len(rqs)))
+  logging.warn("Attempting to write results to {}{}".format(out, ".gz" if zp else ""))
+  if any(rqs) or any(sqs) or out:
+    if zp:
+      with gzip.GzipFile(out + '.gz', 'w') as fh:
+        fh.write(qc.JsonSerialisable.ToJson(mzqc_assembly(rqs, sqs, out), readability=1).encode('utf-8'))
+    else:
+      with open(out, 'w') as fh:
+        fh.write(qc.JsonSerialisable.ToJson(mzqc_assembly(rqs, sqs, out), readability=0))
+  logging.warn("Done. Thank you for choosing QCCalculator!")
+
+
+def common_options(function):
+  function = click.option('--mzml')(function)
+  function = click.option('--output', required=True, type=click.Path(), default="out.mzQC",
+                          help="The path and name of the desired output file.")(function)
+  function = click.option('--zip', default=False,
+                          help="Apply gzip to the output. Appends '.gz' to the target filename and pretty formatting.",
+                          is_flag=True)(function)
+  return function
+
+
+@click.command('basic', short_help='Compute Qc metrics for an mzML file')
+@common_options
+def basic(mzml, output, zip):
+  """
+  Calculate the basic metrics available from virtually every mzML file.
+
+  Parameters
+  ----------
+  mzml: compute the Qc metrics for a mzML file
+
+  Returns
+  -------
+
+  """
+  exp = oms.MSExperiment()
+  oms.MzMLFile().load(click.format_filename(mzml), exp)
+  rq = basicqc.getBasicQuality(exp)
+  rqs.append(rq)
+
+  global_variables(output, zip)
+  finale()
+
+
+@click.command("full", short_help='Compute the QC metrics for spectra file (mzML) and identification data')
+@click.option('--mzid', type=click.Path(exists=True),
+              help="If you have a corresponding mzid file you need to pass it, too. Mutually exclusive to idxml.")
+@click.option('--idxml', type=click.Path(exists=True),
+              help="If you have a corresponding idxml file you need to pass it, too. Mutually exclusive to mzid.")
+@common_options
+def full(mzid=None, idxml=None, mzml=None, output=None, zip=None):
```

Comment:
When you call `qccalculator basic` or `qccalculator full` I think it would be good if you'd essentially get the qccalculator command help info, however right now you get instead:
```
qccalculator full
Traceback (most recent call last):
  File "/usr/local/bin/qccalculator", line 11, in <module>
    load_entry_point('qccalculator', 'console_scripts', 'qccalculator')()
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/workspaces/qccalculator/qccalculator/cli.py", line 151, in full
    oms.MzMLFile().load(click.format_filename(mzml), exp)
  File "/usr/local/lib/python3.6/site-packages/click/utils.py", line 377, in format_filename
    return filename_to_ui(filename)
  File "/usr/local/lib/python3.6/site-packages/click/_compat.py", line 474, in filename_to_ui
    value = value.encode("utf-8", "surrogateescape").decode("utf-8", "replace")
AttributeError: 'NoneType' object has no attribute 'encode'
```
The maxq command already has the defaults set I think. Can you change it so that no arguments default to help or missing arguments?

Reply:
problem solved